### PR TITLE
Add max desired load, remove event waiting

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

File: go/autoscaler/autoscaler.go [hunk 1]
```diff
@@ -213,7 +214,30 @@ nextJob:
 	return js
 }
 
-func scaleDryRun(r *ClusterResource, j job, curDiff int, scaleDown bool) (additional int) {
+func scaleDownDryRun(r *ClusterResource, j job, curDiff int, maxLoadDesired float64) (additional int) {
```
Comment (helinwang) [inline go/autoscaler/autoscaler.go]: Thanks! Two functions is great!

File: go/autoscaler/autoscaler.go [hunk 2]
```diff
@@ -391,27 +403,18 @@ func (a *Autoscaler) Monitor() {
...
-					log.Error(
-						"Error getting the trainer k8s job, retry if have retry remaining",
-						"name", e.Job.ObjectMeta.Name,
-						"retry remaining", retry,
-						"error", err,
-					)
-					time.Sleep(3 * time.Second)
-				}
-
+				tj, err = a.cluster.GetTrainerJob(e.Job)
```
Comment (helinwang) [inline go/autoscaler/autoscaler.go]: 👍 deleting code is awesome!

File: go/cmd/controller/controller.go [hunk 3]
```diff
@@ -48,7 +50,10 @@ func main() {
 	}
 
 	cluster := controller.NewCluster(clientset)
-	as := autoscaler.New(cluster)
+
+	as := autoscaler.New(cluster,
+		func(as *autoscaler.Autoscaler) { as.maxLoadDesired = maxLoadDesired })
```
Comment (helinwang) [inline go/cmd/controller/controller.go]: Does this line compile? `maxLoadDesired` is not visible outside the package.

One way is to add this function inside the `autoscaler` package:

```go
package autoscaler

func WithMaxLoad(maxLoad float64) func(a *Autoscaler) {
  a. maxLoadDesired = maxLoad
}
```

Comment (typhoonzero) [reply go/cmd/controller/controller.go]: No. Will update soon to fix this.

**Pull request merged**